packages/nx/src/command-line/run/run-one.ts

```diff
@@ -84,27 +87,62 @@ export async function runOne(
       { nxJson },
       nxArgs,
       overrides,
-      opts.project,
+      projectName,
       extraTargetDependencies,
       extraOptions
     );
     process.exit(status);
   }
 }
 
-function getProjects(projectGraph: ProjectGraph, project: string): any {
-  if (!projectGraph.nodes[project]) {
-    output.error({
-      title: `Cannot find project '${project}'`,
-    });
-    process.exit(1);
+function getProjects(
```

Comment:
What if this function used `findMatchingProjects` and then checks if the resulting matching projects is a set of a single project? :eyes:

Reply:
Done in both `run-one.ts` and `project.ts`